### ISSUE TITLE
[Haskell] Linter errors are not reported if the file is inside a multi-package stack project

### ISSUE THREAD

**Issue by marchgithub** [dense-analysis/ale#2789]
## Information

**VIM version**

NVIM v0.3.4
Build type: Release

Operating System: NixOS 19.03

## What went wrong

Linter errors are not reported if the file is inside a multi-package stack project. This is because `stack build`

## Reproducing the bug

1. Lint a file inside a multi-package stack project.
2. Errors are found by GHC, but the `stack build` output is incorrectly parsed by the ALE haskell handler.

### :ALEInfo

```
 Current Filetype: haskell
Available Linters: ['cabal_ghc', 'ghc', 'ghc_mod', 'hdevtools', 'hie', 'hlint', 'stack_build', 'stack_ghc']
   Linter Aliases:
'cabal_ghc' -> ['cabal-ghc']
'ghc_mod' -> ['ghc-mod']
'stack_build' -> ['stack-build']
'stack_ghc' -> ['stack-ghc']
  Enabled Linters: ['stack_build']
 Suggested Fixers: 
  'brittany' - Fix Haskell files with brittany.
  'hfmt' - Fix Haskell files with hfmt.
  'hlint' - Refactor Haskell files with hlint.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'stylish-haskell' - Refactor Haskell files with stylish-haskell.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 0
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'never'
let g:ale_linter_aliases = {}
let g:ale_linters = {'haskell': ['stack_build'], 'python': ['pyls', 'mypy']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(started) ['/run/current-system/sw/bin/bash', '-c', 'stack build --fast']
(finished - exit code 1) ['/run/current-system/sw/bin/bash', '-c', 'stack build --fast']

<<<OUTPUT STARTS>>>
Building all executables for `simple-server' once. After a successful build of all of them, only specified executables will be rebuilt.
simple-server> build (lib + exe)
simple-server> Preprocessing library for simple-server-0.1.0.0..
simple-server> Building library for simple-server-0.1.0.0..
simple-server> Preprocessing executable 'simple-server-exe' for simple-server-0.1.0.0..
simple-server> Building executable 'simple-server-exe' for simple-server-0.1.0.0..
simple-server> [1 of 2] Compiling Main
simple-server> 
simple-server> /home/nitori/repos/simple-server/simple-server/app/Main.hs:45:5: error:
simple-server>     Variable not in scope:
simple-server>       eeeeeeresponseLBS
simple-server>         :: Status -> [(HeaderName, [Char])] -> [Char] -> Response
simple-server>    |
simple-server> 45 |     eeeeeeresponseLBS status200 [(hContentType, "text/html")] "Hello"
simple-server>    |     ^^^^^^^^^^^^^^^^^

--  While building package simple-server-0.1.0.0 using:
      /home/nitori/.stack/setup-exe-cache/x86_64-linux-nix/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-linux-nix/Cabal-2.4.0.1 build lib:simple-server exe:simple-server-exe --ghc-options ""
    Process exited with code: ExitFailure 1
<<<OUTPUT ENDS>>>
```

## Suggested fix

For the people who looked it up online, just add `--no-interleaved-output` to `ale_haskell_stack_build_options`. This is a trivial fix on the user side, but it would be nice if the regex for the GHC output handler accounted for this so that the `stack_build` linter just works.
